compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -914,9 +914,12 @@ impl<'a> Linker for MsvcLinker<'a> {
         self.cmd.arg("/guard:cf");
     }
 
-    fn debuginfo(&mut self, strip: Strip, natvis_debugger_visualizers: &[PathBuf]) {
-        match strip {
-            Strip::None => {
+    fn debuginfo(&mut self, _strip: Strip, natvis_debugger_visualizers: &[PathBuf]) {
+        match self.sess.split_debuginfo() {
+            SplitDebuginfo::Off => {
+                self.cmd.arg("/DEBUG:NONE");
```

Comment:
`-Csplit-debuginfo=off` is supposed to put all debuginfo in the executable itself. This isn't possible for MSVC, right? And it is definitively not what `/DEBUG:NONE` does. `-Csplit-debuginfo=unpacked` is supposed to keep all debuginfo in their original object files, which `/DEBUG:FASTLINK` does AFAIU.

Reply:
> `-Csplit-debuginfo=off` is supposed to put all debuginfo in the executable itself.

In other words, if you use `-Csplit-debuginfo=off` only the executable itself is necessary for debugging. With `-Csplit-debuginfo=packed`, the executable and a single separate file containing the debuginfo is necessary. And with `-Csplit-debuginfo=unpacked` all build artifacts may be necessary for debugging.

Reply:
The codegen options documentation already states that `split-debuginfo=off` is supported by all platforms except MSVC. Doesn't this imply that if the platform doesn't support including debug information in the executable, it won't generate any debug information?

Reply:
It shouldn't have been added to macOS either for the same reason as it shouldn't be added to MSVC.

Reply:
I was wondering if I would need another MCP and approval to add `split-debuginfo=unpacked` as `/DEBUG:FASTLINK`.

Reply:
I am wondering if I need another MCP and approval to add `split-debuginfo=unpacked` as `/DEBUG:FASTLINK`.

Reply:
What do you think about not generating PDB files if `debuginfo=none` instead? AFAIK, DWARF objects are omitted when `debuginfo=none`.

Reply:
What do you think about not generating PDB files if `debuginfo=none` instead? AFAIU, DWARF objects are omitted when `debuginfo=none`.